ts/main.ts

```diff
@@ -63,38 +76,99 @@ $form.addEventListener('submit', (event: Event) => {
+      entryId: data.nextEntryId,
+    };
+    data.nextEntryId = data.nextEntryId + 1;
 
-  data.nextEntryId = data.nextEntryId + 1;
+    // every time we hit submit, a DOM tree is created or one entry li and appending it to ul a list
+    // of entries
+    // adding new post data to the top or beginning, so we know its a new post
+    data.entries.unshift(entry);
 
-  // every time we hit submit, a DOM tree is created or one entry li and appending it to ul a list
-  // of entries
-  // so its created quickly after submit which solves the issue of refreshing
-  // adding the new post data to the top or beginning, so we know its a new post
-  data.entries.unshift(entry);
+    // adding only the newly created entry not the whole entries so we can see it in DOM
+    // prepend adds the new li at the beginning not at the end which is similar to unshift()
+    // to maintain the same order in the DOM or html file like in the array entries
+    $ulList.prepend(renderEntry(entry));
 
-  // adding only the newly created entry not the whole entries
-  // prepend adds the new li at the beginning not at the end which is similar to unshift()
-  // to maintain the same order in the DOM or html file like in the array entries
-  $ulList.prepend(renderEntry(entry));
+    // adding new entry data to storage
+    writeData();
+    // reset image
+    $img.src = 'images/placeholder-image-square.jpg';
+    $form.reset();
+    // when the form reset it immediately goes to entries
+    viewSwap('entries');
+    toggleNoEntries();
+  }
 
-  writeData();
-  $img.src = 'images/placeholder-image-square.jpg';
-  $form.reset();
-  // when the form reset it immediately goes to entries
-  viewSwap('entries');
-  toggleNoEntries();
+  // when data.editing array has data to edit
+  else if (data.editing !== null) {
+    // we only change the entry id
+    // so we can use it in comparison
+    const entry: Entry = {
+      entryTitle: formTitle.value,
+      entryId: data.editing.entryId,
+      entryTextArea: formTextArea.value,
+      entryURL: formURL.value,
+    };
+
+    // finding the original entry in data.entries, and then replacing it with the new entry in line 119
+    // so we can then append it to the ul
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === entry.entryId) {
+        data.entries[i] = entry;
+      }
+    }
+
+    // capturing all li's in the Ul
+    const $liElements = document.querySelectorAll('li');
+
+    for (let i = 0; i < $liElements.length; i++) {
+      if (Number($liElements[i].dataset.entryId) === entry.entryId) {
+        const newLi = renderEntry(entry);
+        $liElements[i].replaceWith(newLi);
+        const $newH2 = document.createElement('h2');
+        $newH2.setAttribute('class', 'updated');
+        // trying to append H2 to old LI that is replaced
+        // use the newLi rendered
+        newLi.appendChild($newH2);
+        $newH2.textContent = 'entry is Updated';
```

Comment:
Delete this, we shouldn't see this. This isn't shown in the wireframes or in the instructions